templates/admin/product/box/general.php

```diff
@@ -10,12 +10,16 @@
 ?>
 <fieldset>
 	<?php
+	$regular_price = 0;
+	if ($product instanceof Product\Simple || $product instanceof Product\Virtual || $product instanceof Product\External) {
+		$regular_price = $product->getRegularPrice();
+	}
 	Forms::text(array(
 		'name' => 'product[regular_price]',
 		'label' => __('Price', 'jigoshop').' ('.Currency::symbol().')',
 		'placeholder' => __('Price not announced', 'jigoshop'),
 		'classes' => array('product-simple', (!($product instanceof Product\Variable)) ? '' : 'not-active'),
-		'value' => $product instanceof Product\Simple ? $product->getRegularPrice() : 0,
+		'value' => $regular_price,
```

Comment:
I think this will be better:
`'classes' => array('product-simple', ($product instanceof Product\Purchasable) ? '' : 'not-active'),
'value' => $product instanceof Product\Purchasable ? $product->getRegularPrice() : 0,`